torch/_subclasses/fake_tensor.py

```diff
@@ -444,8 +453,25 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
             else:
                 return args[0].fake_device
 
+        with enable_torch_dispatch_mode(self), torch.overrides.enable_reentrant_dispatch():
+            if func in meta_table:
+                r = meta_table[func](*args, **kwargs)
+                return r
+            elif func in decomposition_table:
+                r = decomposition_table[func](*args, **kwargs)
+                return r
```

Comment:
I'm not sure if we necessarily want this one by default. Tell me more.

Reply:
Well, for operators that have decompositions but no meta-functions, how should we handle them?

I'd be fine with adding a kwarg to `FakeTensor` and branching on whether we're tracing with symbolic shapes (I think that's needed anyways for deciding whether to use C++ meta functions?).

Reply:
The general strategy is that we go to the meta function, and if there's a decomp that got registered to the Meta key so it works. Which suggests this shouldn't even be necessary?

Reply:
The issue here is that we're not generally willing to go to the Meta C++ dispatch key, since we don't know if a decomposition, a python-meta function, or a C++ meta function is registered to it.

I guess I'd be fine with perhaps adding `register_meta` to all `register_decomposition` calls?

Reply:
and the hit the meta table, sgtm